### PR TITLE
viosock: Remove Coinstaller Stuff from the INF File

### PR DESCRIPTION
**Pull request by MartinDrab**
The driver is installed in the Driver Store (DIRID 13) instead of the System Drivers directory (DIRID 12).

### CODE REVIEW

**Comment by xuehuihui** [comment]
Is the protocol not registered?

**Comment by xuehuihui** [comment]
[VirtioSocket_Device_CoInstaller_AddReg]
HKR,,CoInstallers32,0x00010000,"viosocklib.dll,ViosockCoInstaller"
==>
The old INFs  will register the protocol through the ViosockCoInstaller interface.

**Comment by MartinDrab** [comment]
> [VirtioSocket_Device_CoInstaller_AddReg] HKR,,CoInstallers32,0x00010000,"viosocklib.dll,ViosockCoInstaller" ==> The old INFs will register the protocol through the ViosockCoInstaller interface.

Ah, OK, I missed that. I will improve the PR.

File: viosock/sys/viosock.inx [hunk 1]
```diff
@@ -26,7 +26,7 @@ DriverPackageDisplayName = %VirtioSocket.DeviceDesc%
 PnpLockdown = 1
 
 [DestinationDirs]
-DefaultDestDir = 12
+DefaultDestDir = 13
 VirtioSocket_Lib_CopyFiles = 11
```
Comment (kostyanf14) [inline viosock/sys/viosock.inx:30]: According to https://learn.microsoft.com/en-us/windows-hardware/drivers/develop/run-from-driver-store#other-files, you should also move VirtioSocket_Lib_CopyFiles to 13. Please check if is this possible or not. This does not work for RNG providers.

Comment (MartinDrab) [reply viosock/sys/viosock.inx:30]: I will look into it, thanks. Yes, I know about RNG.

Comment (kostyanf14) [reply viosock/sys/viosock.inx:30]: @MartinDrab Please look into changes in https://github.com/virtio-win/kvm-guest-drivers-windows/pull/1075
HLK for 2022 and attestation signing fails with DIRID13 and without TargetOS decoration

File: viosock/sys/viosock.inx [hunk 2]
```diff
@@ -56,6 +55,15 @@ WdfCoInstaller$KMDFCOINSTALLERVERSION$.dll = 1 ; make sure the number matches wi
 [VirtioSocket_Device.NT]
 CopyFiles=Drivers_Dir,VirtioSocket_Lib_CopyFiles
 
+ [VirtioSocket_Device.10.0.15063.Software]
+ AddSoftware=WSKWSP,,VirtioSocket_Software
```
Comment (kostyanf14) [inline viosock/sys/viosock.inx:59]: 1. How do you expect to sign this driver for Windows Server 2016? After applying a Cumulative Update for Windows Server 2016 for x64-based Systems (KB4598243), the latest OS Build is 14393.4169.

2. According to https://learn.microsoft.com/en-us/windows-hardware/drivers/install/inf-addsoftware-directive: `Each AddSoftware directive describes the installation of standalone software. This directive should be used in an INF file of the **SoftwareComponent** setup class. `
How will this work in this INF with `Class=System`?

File: viosock/sys/viosock.inx [hunk 3]
```diff
@@ -56,6 +55,15 @@ WdfCoInstaller$KMDFCOINSTALLERVERSION$.dll = 1 ; make sure the number matches wi
...
+
+ [VirtioSocket_Software]
+ SoftwareType=1
+ SoftwareBinary=viosocklib-test.exe
+ SoftwareArguments="/i"
+ SoftwareVersion=1.0.0.0
```
Comment (kostyanf14) [inline viosock/sys/viosock.inx:65]: Should this version be equal to the driver version or the binary version?

File: viosock/sys/viosock.inx (hunk 2 above)
Comment (MartinDrab) [reply viosock/sys/viosock.inx:59]: It seems that the `SoftwareComponent` class is not a requirement. For example, INF for NVDA Display Driver on my machine uses the `AddSoftware` directive but is of class `Display`.

Comment (MartinDrab) [reply viosock/sys/viosock.inx:59]: 1) well, I suppose there need to be special INF sections for handling OS versions that do not support the `AddSoftware` directive. I am currently experimenting with this.
Unfortunately, it is not possible to register the Winsock Provider by just adding certain keys and values into the Registry (like in `viorng` case); the `AddReg` directive is not strong enough for that.

**Comment by kostyanf14** [comment]
@MartinDrab 

The build failed

```
C:\EWDK11\Program Files\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(436,5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|Win32. [C:\workspace\VirtIO-EWDK-11-21H2-SDV\viosock\installer\viosock-installer.vcxproj]
```

**Comment by YanVugenfirer** [comment]
@MartinDrab https://learn.microsoft.com/en-us/windows-hardware/drivers/install/using-an-extension-inf-file did you try to check "Extension INFs"?

**Comment by MartinDrab** [comment]
> @MartinDrab https://learn.microsoft.com/en-us/windows-hardware/drivers/install/using-an-extension-inf-file did you try to check "Extension INFs"?

I came across them several days ago when working on another (unrelated) issue. I did not have time to look at them in more detail and possibly use them, however, I hope to get to it shortly.

**Comment by kostyanf14** [comment]
@MartinDrab 
We merge https://github.com/virtio-win/kvm-guest-drivers-windows/pull/1087 to switch to the new EWDK. For now, we just disable viosock build for Win11.

**Comment by MartinDrab** [comment]
> @MartinDrab We merge #1087 to switch to the new EWDK. For now, we just disable viosock build for Win11.

OK. I hope to work on this PR shortly and finish it. I apologize for this inconvenience.

**Comment by kostyanf14** [comment]
> > @MartinDrab We merge #1087 to switch to the new EWDK. For now, we just disable viosock build for Win11.
> 
> OK. I hope to work on this PR shortly and finish it. I apologize for this inconvenience.

Not a problem. Disabling the build is not a complicated task. This is more problem for companies that release viosock.

**Comment by MartinDrab** [comment]
> > > @MartinDrab We merge #1087 to switch to the new EWDK. For now, we just disable viosock build for Win11.
> > 
> > 
> > OK. I hope to work on this PR shortly and finish it. I apologize for this inconvenience.
> 
> Not a problem. Disabling the build is not a complicated task. This is more problem for companies that release viosock.

Hello,

I hope I overcame issues regarding the co-installer and the socket WSP installation from the INF file. In the end, I decided to separate the WSP installation into a special service since it is possible to install and start it from the INF file. Other possibilities seem to be problematic:
- WDK complains about the co-installer even when it is used only on old versions of Windows 10,
- the `AddSoftware` directive is not supported on old versions of Windows 10.

I hope this should finally pass the tests and build successfully also with new (E)WDKs.

**Comment by kostyanf14** [comment]
@MartinDrab 

Thanks for your work. 
Unfortunately, the build failed again. 

Please also revert https://github.com/virtio-win/kvm-guest-drivers-windows/commit/f7646006430f40014373cc747bb75a0d7f2cf1c2 and update buildAll.bat (https://github.com/virtio-win/kvm-guest-drivers-windows/commit/ad1aed4601ba937625263c54720fbf316685a3e5 disabled Win11 for viosock)

```
add rem NO WIN11 build for viosock for now
if errorlevel 1 goto :fail
call tools\build.bat viosock\sys\viosock.vcxproj "Win10_SDV Win11_SDV" %*
if errorlevel 1 goto :fail
call tools\build.bat viosock\wsk\wsk.vcxproj "Win10_SDV Win11_SDV" %*
if errorlevel 1 goto :fail
call tools\build.bat viosock\viosock-wsk-test\viosock-wsk-test.vcxproj "Win10_SDV Win11_SDV" %*
```

**Review by YanVugenfirer — APPROVED** [review]

**Pull request merged**